Extensions/TweenBehavior/JsExtension.js

```diff
@@ -0,0 +1,267 @@
+        "Matthias Meike",
+        "Open source (MIT License)"
+      )
+      .setExtensionHelpPath("/all-features/tween");
+
+    var tweenBehavior = new gd.BehaviorJsImplementation();
+
+    tweenBehavior.updateProperty = function (
+      behaviorContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === "My first property") {
+        behaviorContent.property1 = newValue;
+        return true;
+      }
+      if (propertyName === "My other property") {
+        behaviorContent.property2 = newValue === "1";
+        return true;
+      }
+
+      return false;
+    };
+
+    tweenBehavior.getProperties = function (behaviorContent) {
+      var behaviorProperties = new gd.MapStringPropertyDescriptor();
+
+      behaviorProperties.set(
+        "My first property",
+        new gd.PropertyDescriptor(behaviorContent.property1)
+      );
+      behaviorProperties.set(
+        "My other property",
+        new gd.PropertyDescriptor(
+          behaviorContent.property2 ? "true" : "false"
+        ).setType("Boolean")
+      );
+
+      return behaviorProperties;
+    };
+
+    tweenBehavior.setRawJSONContent(
+      JSON.stringify({
+        property1: "Initial value 1",
+        property2: true
+      })
+    );
+
+    extension
+      .addBehavior(
+        "TweenBehavior",
+        t("Tween animation"),
+        "TweenBehavior",
+        t("Tween animation"),
+        "",
+        "CppPlatform/Extensions/topdownmovementicon.png",
+        "TweenBehavior",
+        tweenBehavior,
+        new gd.BehaviorsSharedData()
+      )
+      .setIncludeFile("Extensions/TweenBehavior/tweenruntimebehavior.js")
+      .addIncludeFile("Extensions/TweenBehavior/shifty.js");
+    //.addIncludeFile("Extensions/TweenBehavior/LICENSE-MIT.txt");
+
+    // Behavior related
+    extension
+      .addAction(
+        "AddObjectVariableTween",
+        t("Add object variable tween"),
+        t("Add a tween animation for an object variable."),
+        t(
+          "Tween variable _PARAM3_ of _PARAM0_ from _PARAM4_ to _PARAM5_ with easing _PARAM6_ over _PARAM7_ms as _PARAM2_"
+        ),
+        t("Tween animation"),
+        "JsPlatform/Extensions/take_screenshot24.png",
+        "JsPlatform/Extensions/take_screenshot32.png"
+      )
+      .addParameter("objectPtr", t("Object"), "", false)
+      .addParameter("behavior", t("Behavior"), "TweenBehavior")
+      .addParameter("string", t("Tween Identifier"), "", false) // TODO: Change into stringlist
+      .addParameter("objectvar", t("Object variable"), "", false)
+      .addParameter("expression", t("From value"), "", false)
+      .addParameter("expression", t("To value"), "", false)
+      .addParameter("string", t("Easing"), "", false)
+      .setDefaultValue("linear")
+      .addParameter("expression", t("Duration"), "", false)
+      .addParameter("yesorno", t("Destroy this object when tween finishes"), "", false)
+      .setDefaultValue("false")
```

Comment:
`.setDefaultValue("no")` ;)

Reply:
Did I get it right that when I add an `operator` parameter, that it automatically calculates the value from the `expression`  declared below and delivers the result to the actual function inside `tweenruntimebehavior.js`?

Reply:
What is your use case here of an operator?

The way it works is a bit more complex:

You add a "operator" parameter:
```
.addParameter("operator", _("Operator"), "", false)
```

You add an "expression" parameter **just after**:
```
.addParameter("operator", _("Operator"), "", false)
.addParameter("expression", _("Value"), "", false)
```

Finally, you set a function name (the setter), a getter function name and a manipulated type ("number" or "string"):
```
.getCodeExtraInformation()
.setFunctionName("setSpeed") // Indicates to GD the function to change the value, as usual.
.setGetter("getSpeed") // Indicates to GD how to get the current value.
.setManipulatedType("number") // Indicates to GD to generate calls with the operator.
```

GDevelop will generate something like: `yourObject.setSpeed(yourObject.getSpeed() + value)`
If it's a behavior (i.e: you have "object" and "behavior" arguments, and declared the action on the behavior), GDevelop will generate something like: `yourObject.getBehavior("yourBehaviorName").setSpeed(yourObject.getBehavior("yourBehaviorName").getSpeed() + value)`

If there are other arguments, they are passed to both the setter and the getter:
`yourObject.getBehavior("yourBehaviorName").setSpeed(yourObject.getBehavior("yourBehaviorName").getSpeed(arg2, arg3) + value, arg2, arg3)`

But this is quite complex and not sure if operator would be useful for you in this context?

Reply:
I thought it could be used to create an ObjectPositionTween with relative values for the destination.
Could be useful for Pokémon like movement from tile to tile where you create a tween with MyObject.X()+32 for example.

Reply:
I think we should collect all this information and put it into [README-extensions.md](https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md)

Reply:
Documentation is quite scarce for this I admit 😬😬 There is some reference deeply hidden here: http://4ian.github.io/GD-Documentation/GDCore%20Documentation/classgd_1_1_events_code_generator.html#adb8a6366a93c7a144ceb608b241a2813

But it's not saying anything about the special case about operator. I'd like to keep such details outside of the main readme if possible, to avoid cluttering the readme and have people afraid of the complexity or missing important sections.

But agree documentation is lacking and this special handling can only be seen by reading extensions or digging into the code. 

> I thought it could be used to create an ObjectPositionTween with relative values for the destination.
> Could be useful for Pokémon like movement from tile to tile where you create a tween with MyObject.X()+32 for example.

True, on the other hand I think it's fair to ask the user which is using the tween extension to always specify the target position/value as an absolute value. Knowing that it's easy to write `MyObject.X() + 500` if you want a relative movement.

Also, because the "setGetter" is a function name that would be called on the behavior, so you would have to redefine functions getObjectX/getObjectY and so on. Also extra parameters would be passed to these getters if you have extra parameters. 

So I would avoid them for now. Simpler for you. And I've seen other visual game engines where you have just "Set (something) to (value)" actions without operators. I think it's a fair hypothesis to assume that people using the tween actions can use expressions like .X(), .Y(), .Angle, etc.